src/lib/utils/thorchain/memo.ts

```diff
@@ -0,0 +1,136 @@
+import { bn } from 'lib/bignumber/bignumber'
+
+import { THORCHAIN_AFFILIATE_NAME } from './constants'
+
+const assertIsValidPool = (pool: string, memo: string) => {
+  if (!pool) throw new Error(`pool is required in memo: ${memo}`)
+}
+
+const assertIsValidAsset = (asset: string, memo: string) => {
+  if (!asset) throw new Error(`asset is required in memo: ${memo}`)
+}
+
+const assertIsValidDestAddr = (destAddr: string, memo: string) => {
+  if (!destAddr) throw new Error(`destination address is required in memo: ${memo}`)
+}
+
+const assertIsValidPairedAddr = (pairedAddr: string, memo: string) => {
+  if (!pairedAddr) throw new Error(`paired address is required in memo: ${memo}`)
+}
+
+const assertIsValidLimit = (limit: string, memo: string) => {
+  const maybeStreamingSwap = limit.match(/\//g)
+  const [lim, interval, quantity] = limit.split('/')
+
+  if (maybeStreamingSwap) {
+    if (maybeStreamingSwap.length !== 3)
+      throw new Error(`invalid streaming parameters in memo: ${memo}`)
+    if (!bn(lim).isInteger()) throw new Error(`limit is required in memo: ${memo}`)
+    if (!bn(interval).isInteger()) throw new Error(`interval is required in memo: ${memo}`)
+    if (!bn(quantity).isInteger()) throw new Error(`quantity is required in memo: ${memo}`)
+  }
+
+  if (!bn(limit).isInteger() || !bn(limit).gt(0))
+    throw new Error(`limit is required in memo: ${memo}`)
```

Comment:
Do we want to say `positive limit is required`, as this could throw when a limit of `0` is given?

Reply:
01977830c7